### DISCUSSION TITLE
Upgrading from v5 to v6

### DISCUSSION THREAD

**Discussion by FaisalBudiono** [spatie/laravel-permission#2532] [Q&A] [answered]
Hi guys, I'm struggling to make a migration file to upgrade from v5 to v6. I think there are some changes in the table itself, but I'm not sure what is change so I don't know how to make the new migration

**Comment by FaisalBudiono** [comment, marked as answer]
Found the issue, I don't think there are any changes in the tables themselves. I found error on my test suite because the old migration using `PermissionRegistrar` to get `pivotRole` and `pivotPermission` field from the config.

To solve the issue, just remove the `PermissionRegistrar` and call the `pivotRole` or `pivotPermission` directly just like in https://github.com/spatie/laravel-permission/blob/30051d6486e20f8de95d415132611512c263d826/database/migrations/create_permission_tables.php.stub#L17-L18

**Reply by drbyte** [reply]
Was there anything in the upgrade instructions that could have been improved? 
https://github.com/spatie/laravel-permission/blob/main/docs/upgrading.md#upgrading-from-v5-to-v6

**Reply by FaisalBudiono** [reply]
In the 3rd point, I think you can explicitly say something like:
> Replace the old migration content with the new migration stub. This replacement is just to make sure the migration will run smoothly in your next fresh migration and don't worry it has no changes in the `roles` and `permission` table.

Since back then I was afraid that v6 upgrade has a new table/field

**Reply by drbyte** [reply]
Thanks. Will make a note about it.
